### PR TITLE
Create a new session data object each time

### PR DESCRIPTION
**Pull request by monotypical**
* Add unit tests for session data defaults
* Change session data defaults to be a factory function that returns an new instance of the defaults object
* Change usage of session data defaults to use the new factory function
* Re-enable the webapp (set `SERVICE_UNAVAILABLE` in the config to `false`)
* Run webapp jest unit tests on PRs
* Revert recent changes to `server.js` relating to shuttering the service
* Remove some unneeded server-side logs

### CODE REVIEW

File: webapp/api/data/session-data-defaults.js [hunk 1]
```diff
@@ -51,3 +51,10 @@ export default {
   'salvage-services': '',
   submittedFiles: []
 };
+
+const getDefaults = () => {
+  // use JSON serialisation to create a deep clone of the defaults object
```
Comment (CorinWilkins) [inline webapp/api/data/session-data-defaults.js:56]: I'd be tempted to define `const defaultData = ...` inside getDefaults. That way we know we're always created a new object and nested objects.

Comment (monotypical) [reply webapp/api/data/session-data-defaults.js]: 👍 changes made in https://github.com/mcagov/droits/pull/223/commits/d54c983294c3dcbcfc81e275b728692c5d449b3a

File: webapp/server.js [hunk 2]
```diff
@@ -58,17 +57,41 @@ let useHttps = process.env.USE_HTTPS || config.USE_HTTPS;
...
+
+// Support for parsing data in POSTs
+app.use(bodyParser.json());
+app.use(
+  bodyParser.urlencoded({
+    extended: true,
+  })
+);
+
+app.set('trust proxy', 1); // needed for secure cookies on heroku
```
Comment (CorinWilkins) [inline webapp/server.js:82]: Are we running on heroku?

File: webapp/api/data/session-data-defaults.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    expect(defaults).toHaveProperty("wreck-description", "")
+    expect(defaults).toHaveProperty("salvage-services", "")
+    expect(defaults).toHaveProperty("submittedFiles", [])
+  })
+
+  it("shouldn't return information saved to the defaults object to a later call", () => {
+    const defaults1 = sessionDataDefaults()
+    defaults1.personal["full-name"] = "Alice"
+    const defaults2 = sessionDataDefaults()
+    expect(defaults2.personal["full-name"]).toBe("")
```
Comment (bnjn-mt) [inline webapp/api/data/session-data-defaults.test.js:46]: I'd like to add an additional expect to check the two sessionDataDefaults objects are not the same:

`expect(defaults1 === defaults2).toBe(false)`

File: webapp/server.js [hunk 4]
```diff
@@ -94,158 +117,78 @@ app.use('/assets/css', (req, res, next) => {
...
+app.use('/report/check-your-answers', function (req, res, next) {
+  res.set(
+    'Cache-Control',
+    'no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0'
+  );
+  next();
+});
+
+app.get(/^([^.]+)$/, function (req, res, next) {
+  if (config.SERVICE_UNAVAILABLE === true) {
```
Comment (CorinWilkins) [inline webapp/server.js:159]: I'd maybe move this if out into the body of this function and only add the routes required for if the service is unavailable and add an else to add the routes if it is live.

File: webapp/server.js (hunk 2 above)
Comment (bnjn-mt) [reply webapp/server.js:82]: Nope. That's left over from when it did.

File: webapp/api/data/session-data-defaults.test.js (hunk 3 above)
Comment (monotypical) [reply webapp/api/data/session-data-defaults.test.js:62]: I've updated this to check recursively in [1aeb63e](https://github.com/mcagov/droits/pull/223/commits/1aeb63e8d9fcc4053482507a5cb89a28280f7eb4)

File: webapp/server.js (hunk 4 above)
Comment (monotypical) [reply webapp/server.js:159]: Done in [ff53064](https://github.com/mcagov/droits/pull/223/commits/ff530642f32973c44b5838c788d95365aea11c69)

**Review by bnjn-mt — APPROVED** [review]
Happy to go with this. Tested report submission locally and it works fine.

**Pull request merged**